src/sinks/util/encoding/mod.rs

```diff
@@ -66,8 +66,14 @@ pub trait EncodingConfiguration<E> {
                             })
                         })
                         .collect::<VecDeque<_>>();
+
+                    // reverse sort so that we delete array elements at the end first rather than
+                    // the start so that any `nulls` at the end are dropped and empty arrays are
+                    // pruned
+                    to_remove.make_contiguous().sort_by(|a, b| b.cmp(a));
```

Comment:
I'm not sure I fully understand this? Isn't `to_remove` here just a String list of the fields in the log event? So sorting this is just going to sort the fields in alphabetical order? 

I'm not seeing how this is relevant to array elements. Clearly I'm missing some context here..

As an aside, because it isn't changed by this PR, but I don't understand why this is collecting a `VecDeque` and not just a `Vec`. Any ideas? I'm not sure why it needed to collect anything at all untill the need to sort the fields came along..

Reply:
Apologies, I should have provided more context here. The idea is by deleting the keys alphanumerically backwards, that arrays will be pruned correctly. For example, if you have incoming event:

```
{ "hostname": "localhost": "foo": [1,2] }
```

And we do `only_fields = ["hostname"]`, then we want to have `foo` deleted. The way that `.keys()` on the log event works is that it'll return `["hostname", "foo.0", "foo.1"]`. When removing elements from an array, if the element is the last one, it will actually remove it, otherwise it'll set it to null to preserve the index of later elements. So we want to remove `foo.1` and then `foo.0`. Once `foo.0` is removed, the pruning behavior will cause it to drop `foo`.

Admittedly this is a bit of an awkward way to do this, but it fits with the existing log event interface.

I'm unsure why it was a `VecDeque` originally. Seemingly it could just be a `Vec` here?

Reply:
I believe it could just be a `Vec`.